nmsg/input.h

```diff
@@ -119,6 +130,29 @@ nmsg_input_open_zmq(void *s);
 nmsg_input_t
 nmsg_input_open_zmq_endpoint(void *zmq_ctx, const char *ep);
 
+/**
+ * Create a Kafka consumer and initialize a new NMSG stream input from it.
+ *
+ * This function is a wrapper for nmsg_input_open_kafka(). Instead of taking an
+ * already initialized Kafka consumer context, it takes an endpoint argument of
+ * format "proto:topic[#partition]@broker[:port][,offset]"
+ *
+ * The value of proto must be either "nmsg" (binary container input) or "json"
+ * (JSON-serialized payloads) and either or both a partition number and offset
+ * value may be optionally supplied.
+ *
+ * \see nmsg_output_open_kafka_endpoint()
+ *
+ * \param[in] addr Kafka endpoint address string
+ *
+ * \param[in] timeout in milliseconds.
+ *
+ * \return Opaque pointer that is NULL on failure or non-NULL on success.
+ */
+
+nmsg_input_t
+nmsg_input_open_kafka_endpoint(const char *addr, int timeout);
```

Comment:
For
(private) nmsg_input_open_kafka -> _nmsg_input_open_kafka_ctx
(private) nmsg_output_open_kafka -> _nmsg_output_open_kafka_ctx
We are not openning a context, but input/output. Format of private functions in input/output.h is _input/_output thus
_input_open_kafka
_output_open_kafka

Reply:
Good point, and I can see where naming gets confusing. The critical part is getting the `nmsg_{input,output}_open_kafka` removed from the public API / exported symbols, and converted to take a typed pointer instead of `void *` arg.

We can discuss the naming offline.